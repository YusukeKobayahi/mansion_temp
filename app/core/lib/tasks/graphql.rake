# frozen_string_literal: true

module GraphqlTask
  extend Rake::DSL

  namespace :graphql do
    namespace :consumer do
      task schema_dump: :environment do
        require 'graphql/rake_task'

        GraphQL::RakeTask.new(
          load_schema: lambda { |_task|
            require File.expand_path('../../app/graphql/consumer/inno_schema', __dir__)
            Consumer::InnoSchema
          },
          directory: './graphql/consumer',
        )
        Rake::Task['graphql:schema:idl'].invoke
      end
    end
  end
end
