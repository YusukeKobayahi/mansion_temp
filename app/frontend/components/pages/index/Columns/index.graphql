fragment MsLibraryIndexIeulColumn on Prefecture {
  category {
    href
    title
    thumbnail
    description
  }
  ids {
    href
    title
    thumbnail
    description
  }
}
