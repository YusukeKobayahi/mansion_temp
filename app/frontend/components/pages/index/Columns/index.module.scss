@import "~/styles/variables";
@import "~/styles/utils";

.container {
  margin-top: 50px;

  @include screenMd {
    margin-top: 10px;
  }
}

.title {
  background-color: #bca565;
  color: #ffffff;
  font-size: 24px;
  font-weight: bold;
  text-align: center;
  line-height: 50px;

  @include screenMd {
    font-size: 20px;
  }
}

.subTitle {
  margin: 20px auto 0;
  text-align: center;

  @include screenMd {
    width: 80%;
    position: relative;
  }

  h3 {
    display: inline-block;
    position: relative;
    font-size: 24px;
    font-weight: bold;
    color: #746742;

    @include screenMd {
      display: block;
      font-size: 20px;
      line-height: 70px;
      text-align: center;
    }

    &::before {
      content: "";
      position: absolute;
      top: 50%;
      left: -30px;
      transform: translateY(-50%);
      width: 20px;
      height: 3px;
      background-color: #746742;

      @include screenMd {
        transform: unset;
        top: unset;
        bottom: 0;
        left: 0;
        border-bottom: 2px solid #746742;
        border-left: 2px solid #746742;
        width: 15px;
        height: 15px;
        display: block;
        background-color: unset;
      }
    }

    &::after {
      content: "";
      position: absolute;
      top: 50%;
      right: -30px;
      transform: translateY(-50%);
      width: 20px;
      height: 3px;
      background-color: #746742;

      @include screenMd {
        transform: unset;
        top: unset;
        content: "";
        position: absolute;
        bottom: 0;
        right: 0;
        border-bottom: 2px solid #746742;
        border-right: 2px solid #746742;
        width: 15px;
        height: 15px;
        display: block;
        background-color: unset;
      }
    }
  }

  &::before {
    @include screenMd {
      content: "";
      position: absolute;
      top: 0;
      left: 0;
      border-top: 2px solid #746742;
      border-left: 2px solid #746742;
      width: 15px;
      height: 15px;
      display: block;
    }
  }

  &::after {
    @include screenMd {
      content: "";
      position: absolute;
      top: 0;
      right: 0;
      border-top: 2px solid #746742;
      border-right: 2px solid #746742;
      width: 15px;
      height: 15px;
      display: block;
    }
  }
}

.slider {
  margin-top: 20px;
  padding: 0 40px;
  position: relative;

  @include screenMd {
    flex-wrap: wrap;
    padding: 0 15px;
  }

  .arrow {
    width: 16px;
    height: 16px;
    position: absolute;
    top: 45%;
    background-color: unset;
    cursor: pointer;

    @include screenMd {
      display: none;
    }

    &.prev {
      border-top: 2px solid #bca565;
      border-left: 2px solid #bca565;
      transform: rotate(-45deg) translateY(-50%);
      left: 10px;
    }

    &.next {
      border-top: 2px solid #bca565;
      border-right: 2px solid #bca565;
      transform: rotate(45deg) translateY(-50%);
      right: 10px;
    }
  }

  &Inner {
    overflow-x: hidden;
  }

  .slides {
    display: flex;
    transition: 0.3s;
    @include screenMd {
      flex-wrap: wrap;
    }
  }

  .block {
    width: 200px;
    min-width: 200px;
    margin-right: 20px;
    border: 1px solid #e5e5e5;
    box-sizing: border-box;

    @include screenMd {
      min-width: 50%;
      width: 50%;
      box-sizing: border-box;
      border: none;
      border-bottom: 1px solid #d8d8d8;
      padding: 9px 0;
      margin: 0;

      &.spHide {
        display: none;
      }
    }

    &:nth-child(odd) {
      @include screenMd {
        padding-right: 9px;
      }
    }

    &:nth-child(even) {
      @include screenMd {
        padding-left: 9px;
        border-left: 1px solid #d8d8d8;
      }
    }

    .image {
      width: 100%;
      height: 115px;
      background-color: #eaeaea;

      img {
        width: 100%;
        height: 100%;
        object-fit: cover;
      }
    }

    .text {
      padding-top: 10px;
      color: #bca565;
      font-size: 14px;
      line-height: 1.5;
      font-weight: bold;
      text-decoration: underline;
      padding: 15px;

      @include screenMd {
        padding: 10px;
      }
    }
  }

  .pagers {
    margin-top: 20px;
    display: flex;
    justify-content: center;

    .pager {
      background-color: unset;
      cursor: pointer;

      span {
        font-size: 24px;
        color: #efe4c5;

        &.current {
          color: #b2a173;
        }
      }
    }

    @include screenMd {
      display: none;
    }
  }
}

.columns {
  margin-top: 40px;
  display: flex;
  flex-wrap: wrap;
  @include screenMd {
    margin-top: 40px;
    padding: 0 15px;
  }

  .column {
    width: calc((100% - 40px) / 3);
    display: flex;
    align-items: center;
    padding: 10px 30px 10px 0;
    box-sizing: border-box;
    border-bottom: 1px solid #d8d8d8;
    position: relative;

    &:nth-child(3n + 1),
    &:nth-child(3n + 2) {
      margin-right: 20px;
      @include screenMd {
        margin-right: 0;
      }
    }

    @include screenMd {
      width: 1000%;
      padding: 0 20px 10px 0;
      margin-bottom: 10px;
    }

    &:nth-last-child(1) {
      margin-bottom: 0;
    }

    .image {
      min-width: 81px;
      width: 81px;
      height: 81px;
      background-color: #eaeaea;
      margin-right: 12px;

      img {
        height: 100%;
        width: 100%;
        object-fit: cover;
      }
    }

    .text {
      color: #bca565;
      font-size: 15px;
      line-height: 1.5;
      font-weight: bold;
      text-decoration: underline;

      @include screenMd {
        font-size: 14px;
      }
    }

    &::after {
      content: "";
      width: 10px;
      height: 10px;
      border-top: 2px solid #bca565;
      border-right: 2px solid #bca565;
      transform: rotate(45deg) translateY(-50%);
      position: absolute;
      right: 18px;
      top: 50%;

      @include screenMd {
        width: 10px;
        height: 10px;
        border-top: 1px solid #bca565;
        border-right: 1px solid #bca565;
        right: 12px;
      }
    }
  }
}

.btn {
  width: 200px;
  border: 2px solid #bca565;
  border-radius: 40px;
  background-color: #f8f6f0;
  margin: 20px auto 0;
  font-size: 20px;
  color: #b2a173;
  line-height: 56px;
  text-align: center;
  font-weight: bold;

  &Wrapper {
    text-align: center;
  }
}
