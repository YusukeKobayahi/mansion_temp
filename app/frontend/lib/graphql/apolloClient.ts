// 参考
// https://github.com/vercel/next.js/tree/canary/examples/with-apollo

import { useMemo } from "react";
import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
import { createHttpLink } from "apollo-link-http";
import { from } from "apollo-link";
import { createFragmentArgumentLink } from "apollo-link-fragment-argument";
import { mergeDeep, relayStylePagination } from "@apollo/client/utilities";
import { isSSR } from "~/lib/utils";

/* eslint-disable @typescript-eslint/no-explicit-any -- キャッシュの型を事前に決められないため */
type CacheState = Record<string, any> | null;
/* eslint-enable @typescript-eslint/no-explicit-any */

/* eslint-disable @typescript-eslint/naming-convention -- GraphQLのタイプ名 (このconventionに従わない)を指定したいので */
const createCache = () =>
  new InMemoryCache({
    typePolicies: {
      Mansion: {
        keyFields: ["uniqueCode"],
        fields: {
          salesHistories: relayStylePagination(),
        },
      },
    },
  });
/* eslint-enable @typescript-eslint/naming-convention */

const link = from([
  createFragmentArgumentLink(),
  createHttpLink({
    uri: `${process.env.NEXT_PUBLIC_CORE_API_URL}/consumer/graphql`,
    fetchOptions: {
      mode: "cors",
    },
  }),
]);

function createApolloClient() {
  return new ApolloClient({
    ssrMode: isSSR(),
    // See: https://github.com/apollographql/apollo-link/issues/1258#issuecomment-604031488
    // TODO: @apollo/clientバージョン上げたら直るかも
    link: (link as unknown) as ApolloLink,
    cache: createCache(),
    defaultOptions: {
      watchQuery: {
        // SSR時にuseQueryによってリクエストしてレンダリングする場合、
        // getServerSidePropsからキャッシュが渡っている必要がある。(でないと描画されない)
        // そのため、SSR時は常にcacheから取得されるべきで、実装意図を明確にするためPolicyも設定しておく
        // (エラーになるわけではない)
        fetchPolicy: isSSR() ? "cache-only" : "cache-first",
      },
    },
  });
}

let apolloClient: ReturnType<typeof createApolloClient>;

export function initializeApollo(
  initialState: CacheState = null
): ApolloClient<CacheState> {
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Merge the existing cache into data passed from getStaticProps/getServerSideProps
    const data = mergeDeep(initialState, existingCache);

    // Restore the cache with the merged data
    _apolloClient.cache.restore(data);
  }
  // For SSG and SSR always create a new Apollo Client
  if (isSSR()) return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: CacheState): ApolloClient<CacheState> {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
