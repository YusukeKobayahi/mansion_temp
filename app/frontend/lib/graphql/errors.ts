export interface GraphQLError {
  graphQLErrors?: {
    message: string;
    extensions?: {
      code: string;
    };
  }[];
}

export const isNotFoundError = (err: GraphQLError): boolean => {
  const graphQLErrors = err.graphQLErrors;
  if (graphQLErrors == null) return false;

  return graphQLErrors[0]?.extensions?.code === "NOT_FOUND";
};
