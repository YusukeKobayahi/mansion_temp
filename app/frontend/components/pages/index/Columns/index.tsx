import { MsLibraryIndexIeulColumnFragment } from "~/graphql/generated";
import styles from "./index.module.scss";
import classnames from "classnames";
import { useState } from "react";
import { buildArrayFromNumber } from "~/lib/utils";
import { event } from "~/lib/gtag";

type ColumnsProps = {
  data: MsLibraryIndexIeulColumnFragment[];
};

const Columns: React.FC<ColumnsProps> = ({ data }: ColumnsProps) => {
  const [currentNum, setCurrentNum] = useState(0);
  const [isVisible, setIsVisible] = useState(false);
  const categoryPosts = data[0].category;
  const idsPosts = data[0].ids;
  if (![categoryPosts, idsPosts].every((el) => el && el.length !== 0))
    return null;
  const clickColumn = () => {
    event({
      action: "msl_content_click",
      category: "msl_top",
      label: "msl_column",
      value: 1,
    });
  };

  const totalCategoryPosts = Number(categoryPosts?.length);
  const spDefaultVisiblePosts = 4;
  const categoryPostsBlocks = categoryPosts?.map((a, i) => {
    return (
      <a
        href={String(a.href)}
        key={i}
        className={classnames(
          styles.block,
          i > spDefaultVisiblePosts - 1 && !isVisible ? styles.spHide : ""
        )}
        onClick={clickColumn}
      >
        <div className={styles.image}>
          <img src={String(a.thumbnail)} alt={String(a.title)} />
        </div>
        <div className={styles.text}>{a.title}</div>
      </a>
    );
  });
  const idsPostsBlocks = idsPosts?.map((a, i) => {
    return (
      <a href={String(a.href)} key={i} className={styles.column}>
        <div className={styles.image}>
          <img src={String(a.thumbnail)} alt={String(a.title)} />
        </div>
        <div className={styles.text}>{a.title}</div>
      </a>
    );
  });
  const slideAmount = {
    width: 220 * totalCategoryPosts,
    transform: `translateX(${-220 * currentNum}px)`,
  };
  const slideNext = () => {
    if (currentNum < totalCategoryPosts - 4) {
      setCurrentNum(currentNum + 1);
    } else {
      setCurrentNum(0);
    }
  };
  const slidePrev = () => {
    if (currentNum > 0) {
      setCurrentNum(currentNum - 1);
    } else {
      setCurrentNum(totalCategoryPosts - 4);
    }
  };
  const pagers = buildArrayFromNumber(totalCategoryPosts - 3).map((i) => {
    return (
      <button className={styles.pager} onClick={() => setCurrentNum(i)} key={i}>
        {currentNum === i ? (
          <span className={styles.current}>●</span>
        ) : (
          <span>●</span>
        )}
      </button>
    );
  });
  return (
    <section className={styles.container}>
      <h2 className={styles.title}>物件探しお役立ちコラム</h2>
      <div className={styles.subTitle}>
        <h3>新着コラム</h3>
      </div>
      <div className={classnames(styles.slider, "pcOnly")}>
        <button
          onClick={slidePrev}
          className={classnames(styles.arrow, styles.prev)}
        ></button>
        <button
          onClick={slideNext}
          className={classnames(styles.arrow, styles.next)}
        ></button>
        <div className={styles.sliderInner}>
          <div className={styles.slides} style={slideAmount}>
            {categoryPostsBlocks}
          </div>
        </div>
        <div className={styles.pagers}>{pagers}</div>
      </div>
      <div className={classnames(styles.slider, "spOnly")}>
        <div className={styles.sliderInner}>
          <div className={styles.slides}>{categoryPostsBlocks}</div>
        </div>
        {!isVisible && (
          <div className={styles.btnWrapper}>
            <button className={styles.btn} onClick={() => setIsVisible(true)}>
              もっと見る
            </button>
          </div>
        )}
      </div>
      <div className={styles.columns}>{idsPostsBlocks}</div>
    </section>
  );
};

export default Columns;
