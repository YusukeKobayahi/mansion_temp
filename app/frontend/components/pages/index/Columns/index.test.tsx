import { render } from "@testing-library/react";
import mockData from "~/components/pages/index/fixtures/mockData.json";
import { IndexQuery } from "~/graphql/generated";
import Columns from "~/components/pages/index/Columns";

describe("components/pages/index/Columns", () => {
  it("正常に表示される", () => {
    const mock = mockData as IndexQuery;
    const { asFragment } = render(<Columns data={mock.prefectures} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
