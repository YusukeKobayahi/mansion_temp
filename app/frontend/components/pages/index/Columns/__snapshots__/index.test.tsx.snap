// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/index/Columns 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2
      class="title"
    >
      物件探しお役立ちコラム
    </h2>
    <div
      class="subTitle"
    >
      <h3>
        新着コラム
      </h3>
    </div>
    <div
      class="slider pcOnly"
    >
      <button
        class="arrow prev"
      />
      <button
        class="arrow next"
      />
      <div
        class="sliderInner"
      >
        <div
          class="slides"
          style="width: 1760px; transform: translateX(0px);"
        >
          <a
            class="block"
            href="/column/articles/21212"
          >
            <div
              class="image"
            >
              <img
                alt="中古マンションの築年数は何年が買い時？寿命や耐震性まで解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/second-hand-mansion-age-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              中古マンションの築年数は何年が買い時？寿命や耐震性まで解説
            </div>
          </a>
          <a
            class="block"
            href="/column/articles/21227"
          >
            <div
              class="image"
            >
              <img
                alt="マンションか戸建てかどっちか決められない！悩んだ時のポイントを解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/築30年のマンション-1-1024x576.png"
              />
            </div>
            <div
              class="text"
            >
              マンションか戸建てかどっちか決められない！悩んだ時のポイントを解説
            </div>
          </a>
          <a
            class="block"
            href="/column/articles/37424"
          >
            <div
              class="image"
            >
              <img
                alt="共働き夫婦が戸建てかマンションの購入で悩んだ時の判断ポイント"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/11/domannian-mansion-joint-work-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              共働き夫婦が戸建てかマンションの購入で悩んだ時の判断ポイント
            </div>
          </a>
          <a
            class="block"
            href="/column/articles/24029"
          >
            <div
              class="image"
            >
              <img
                alt="賃貸か購入、住むならどっち？迷った時の5つのポイントを紹介！"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/06/rent-purchase-which-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              賃貸か購入、住むならどっち？迷った時の5つのポイントを紹介！
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/37425"
          >
            <div
              class="image"
            >
              <img
                alt="40代は「賃貸」「購入」のどちらがおすすめ？考え方のヒント"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/11/rental-and-purchase-40s-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              40代は「賃貸」「購入」のどちらがおすすめ？考え方のヒント
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/37426"
          >
            <div
              class="image"
            >
              <img
                alt="未公開物件の探し方やメリット・デメリットを詳しく解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/11/how-to-find-an-unpublished-property-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              未公開物件の探し方やメリット・デメリットを詳しく解説
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/24049"
          >
            <div
              class="image"
            >
              <img
                alt="未公開物件とは？ネットに載らない物件情報を得る方法や注意点を徹底解説！"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/06/unpublished-property-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              未公開物件とは？ネットに載らない物件情報を得る方法や注意点を徹底解説！
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/372"
          >
            <div
              class="image"
            >
              <img
                alt="マンションの間取りはどう選ぶ？理想の間取りを選ぶポイントを解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2018/10/shutterstock_170805509-e1635758601976.jpg"
              />
            </div>
            <div
              class="text"
            >
              マンションの間取りはどう選ぶ？理想の間取りを選ぶポイントを解説
            </div>
          </a>
        </div>
      </div>
      <div
        class="pagers"
      >
        <button
          class="pager"
        >
          <span
            class="current"
          >
            ●
          </span>
        </button>
        <button
          class="pager"
        >
          <span>
            ●
          </span>
        </button>
        <button
          class="pager"
        >
          <span>
            ●
          </span>
        </button>
        <button
          class="pager"
        >
          <span>
            ●
          </span>
        </button>
        <button
          class="pager"
        >
          <span>
            ●
          </span>
        </button>
      </div>
    </div>
    <div
      class="slider spOnly"
    >
      <div
        class="sliderInner"
      >
        <div
          class="slides"
        >
          <a
            class="block"
            href="/column/articles/21212"
          >
            <div
              class="image"
            >
              <img
                alt="中古マンションの築年数は何年が買い時？寿命や耐震性まで解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/second-hand-mansion-age-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              中古マンションの築年数は何年が買い時？寿命や耐震性まで解説
            </div>
          </a>
          <a
            class="block"
            href="/column/articles/21227"
          >
            <div
              class="image"
            >
              <img
                alt="マンションか戸建てかどっちか決められない！悩んだ時のポイントを解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/築30年のマンション-1-1024x576.png"
              />
            </div>
            <div
              class="text"
            >
              マンションか戸建てかどっちか決められない！悩んだ時のポイントを解説
            </div>
          </a>
          <a
            class="block"
            href="/column/articles/37424"
          >
            <div
              class="image"
            >
              <img
                alt="共働き夫婦が戸建てかマンションの購入で悩んだ時の判断ポイント"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/11/domannian-mansion-joint-work-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              共働き夫婦が戸建てかマンションの購入で悩んだ時の判断ポイント
            </div>
          </a>
          <a
            class="block"
            href="/column/articles/24029"
          >
            <div
              class="image"
            >
              <img
                alt="賃貸か購入、住むならどっち？迷った時の5つのポイントを紹介！"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/06/rent-purchase-which-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              賃貸か購入、住むならどっち？迷った時の5つのポイントを紹介！
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/37425"
          >
            <div
              class="image"
            >
              <img
                alt="40代は「賃貸」「購入」のどちらがおすすめ？考え方のヒント"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/11/rental-and-purchase-40s-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              40代は「賃貸」「購入」のどちらがおすすめ？考え方のヒント
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/37426"
          >
            <div
              class="image"
            >
              <img
                alt="未公開物件の探し方やメリット・デメリットを詳しく解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/11/how-to-find-an-unpublished-property-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              未公開物件の探し方やメリット・デメリットを詳しく解説
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/24049"
          >
            <div
              class="image"
            >
              <img
                alt="未公開物件とは？ネットに載らない物件情報を得る方法や注意点を徹底解説！"
                src="https://stg.ieul.jp/column/wp-content/uploads/2021/06/unpublished-property-1024x683.jpg"
              />
            </div>
            <div
              class="text"
            >
              未公開物件とは？ネットに載らない物件情報を得る方法や注意点を徹底解説！
            </div>
          </a>
          <a
            class="block spHide"
            href="/column/articles/372"
          >
            <div
              class="image"
            >
              <img
                alt="マンションの間取りはどう選ぶ？理想の間取りを選ぶポイントを解説"
                src="https://stg.ieul.jp/column/wp-content/uploads/2018/10/shutterstock_170805509-e1635758601976.jpg"
              />
            </div>
            <div
              class="text"
            >
              マンションの間取りはどう選ぶ？理想の間取りを選ぶポイントを解説
            </div>
          </a>
        </div>
      </div>
      <div
        class="btnWrapper"
      >
        <button
          class="btn"
        >
          もっと見る
        </button>
      </div>
    </div>
    <div
      class="columns"
    >
      <a
        class="column"
        href="/column/articles/24049"
      >
        <div
          class="image"
        >
          <img
            alt="未公開物件とは？ネットに載らない物件情報を得る方法や注意点を徹底解説！"
            src="https://stg.ieul.jp/column/wp-content/uploads/2021/06/unpublished-property-1024x683.jpg"
          />
        </div>
        <div
          class="text"
        >
          未公開物件とは？ネットに載らない物件情報を得る方法や注意点を徹底解説！
        </div>
      </a>
      <a
        class="column"
        href="/column/articles/21227"
      >
        <div
          class="image"
        >
          <img
            alt="マンションか戸建てかどっちか決められない！悩んだ時のポイントを解説"
            src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/築30年のマンション-1-1024x576.png"
          />
        </div>
        <div
          class="text"
        >
          マンションか戸建てかどっちか決められない！悩んだ時のポイントを解説
        </div>
      </a>
      <a
        class="column"
        href="/column/articles/21212"
      >
        <div
          class="image"
        >
          <img
            alt="中古マンションの築年数は何年が買い時？寿命や耐震性まで解説"
            src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/second-hand-mansion-age-1024x683.jpg"
          />
        </div>
        <div
          class="text"
        >
          中古マンションの築年数は何年が買い時？寿命や耐震性まで解説
        </div>
      </a>
      <a
        class="column"
        href="/column/articles/21224"
      >
        <div
          class="image"
        >
          <img
            alt="マンション買うなら新築と中古どっちがお得？価格差やメリットとデメリットを比較！"
            src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/mansion-newly-and-used-1024x683.jpg"
          />
        </div>
        <div
          class="text"
        >
          マンション買うなら新築と中古どっちがお得？価格差やメリットとデメリットを比較！
        </div>
      </a>
      <a
        class="column"
        href="/column/articles/21197"
      >
        <div
          class="image"
        >
          <img
            alt="マンション購入の注意点を解説！物件選びから内覧・契約時に注意するべきことは？"
            src="https://stg.ieul.jp/column/wp-content/uploads/2021/04/second-hand-mansion-important-1024x683.jpg"
          />
        </div>
        <div
          class="text"
        >
          マンション購入の注意点を解説！物件選びから内覧・契約時に注意するべきことは？
        </div>
      </a>
      <a
        class="column"
        href="/column/articles/372"
      >
        <div
          class="image"
        >
          <img
            alt="マンションの間取りはどう選ぶ？理想の間取りを選ぶポイントを解説"
            src="https://stg.ieul.jp/column/wp-content/uploads/2018/10/shutterstock_170805509-e1635758601976.jpg"
          />
        </div>
        <div
          class="text"
        >
          マンションの間取りはどう選ぶ？理想の間取りを選ぶポイントを解説
        </div>
      </a>
    </div>
  </section>
</DocumentFragment>
`;
